### PR TITLE
Support ruby 3.1

### PR DESCRIPTION
**Pull request by RenzoMinelli**
This PR is based on #178 (Author: Brendan Mulholland), these changes are required for this gem to support Ruby 3.1:

* Update rubocop to 1.22.x
* Update rubocop-rspec to 2.0.x

### CODE REVIEW

File: rubocop-airbnb/spec/rubocop/cop/airbnb/spec_constant_assignment_spec.rb
Comment (RenzoMinelli) [inline rubocop-airbnb/spec/rubocop/cop/airbnb/spec_constant_assignment_spec.rb:19]: The test's description was changed to reflect what it is being tested.

File: rubocop-airbnb/spec/rubocop/cop/airbnb/module_method_in_wrong_file_spec.rb
Comment (RenzoMinelli) [inline rubocop-airbnb/spec/rubocop/cop/airbnb/module_method_in_wrong_file_spec.rb:77]: This test was set as pending due to the fact that a bug is causing it to fail, suggesting baz to be defined in foo/foo.rb instead of being in foo.rb.

Comment (pariser) [reply rubocop-airbnb/spec/rubocop/cop/airbnb/module_method_in_wrong_file_spec.rb:77]: This seems strange to me. Why is the metaclass of `Foo` expected to be defined in `foo/foo.rb`? Was this failing previously? I'm assuming this is due to the rubocop upgrade?

I'd like to dig into this one and figure out why this changed

File: rubocop-airbnb/spec/rubocop/cop/airbnb/spec_constant_assignment_spec.rb
Comment (pariser) [reply rubocop-airbnb/spec/rubocop/cop/airbnb/spec_constant_assignment_spec.rb:19]: Thanks! Good correction

**Review by pariser — COMMENTED** [review]
Just a few minor questions – thanks for the contibution!

File: rubocop-airbnb/spec/rubocop/cop/airbnb/module_method_in_wrong_file_spec.rb
Comment (RenzoMinelli) [reply rubocop-airbnb/spec/rubocop/cop/airbnb/module_method_in_wrong_file_spec.rb:77]: Oh sorry I should have left the link to the [issue](https://github.com/airbnb/ruby/issues/190) that I created discussing this. It's a false positive in master.

File: rubocop-airbnb/config/rubocop-rspec.yml
Comment (RenzoMinelli) [reply rubocop-airbnb/config/rubocop-rspec.yml:59]: The [upgrade guide to version 2.x of rubocop-rspec](https://github.com/rubocop/rubocop-rspec/blob/v2.0.0/docs/modules/ROOT/pages/upgrade_to_version_2.adoc) indicates that the `CustomIncludeMethods` configuration option for `RSpec/EmptyExampleGroup` was removed.

File: rubocop-airbnb/lib/rubocop/cop/airbnb/spec_constant_assignment.rb
Comment (RenzoMinelli) [reply rubocop-airbnb/lib/rubocop/cop/airbnb/spec_constant_assignment.rb]: Thanks, I followed your suggestion in f4562a16336c3517de33a071e3febf90dd2992d1.

File: rubocop-airbnb/config/rubocop-rspec.yml
Comment (pariser) [reply rubocop-airbnb/config/rubocop-rspec.yml:59]: Ah, I must have pulled up an old reference to the `EmptyExampleGroup` docs. Thanks for the reference.

File: rubocop-airbnb/rubocop-airbnb.gemspec
Comment (RenzoMinelli) [inline rubocop-airbnb/rubocop-airbnb.gemspec:28]: I decided to bump the version to grater than 1.22.0 because if I tried running rubocop with any version between 1.0.0 and 1.21.0 errors would occur, indicating the following message: 

```
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
https://github.com/rubocop/rubocop/issues
```

**Comment by pariser** [comment]
@RenzoMinelli Thanks! This is great and I'm pretty sure it's ready to go, but I'd love to get tests passing on this branch before we merge. I just merged https://github.com/airbnb/ruby/pull/191 which switched out travis for github actions (apparently Travis hadn't been working for quite a long time). Would you mind rebasing? You'll have to add ruby 3.1 (and uncomment 3.0) in the CI matrix [here](https://github.com/airbnb/ruby/blob/baee021e39a70924a973bb9a6b8c092b2d8afb9e/.github/workflows/rspec_rubocop.yml#L19-L33).

**Comment by RenzoMinelli** [comment]
> @RenzoMinelli Thanks! This is great and I'm pretty sure it's ready to go, but I'd love to get tests passing on this branch before we merge. I just merged #191 which switched out travis for github actions (apparently Travis hadn't been working for quite a long time). Would you mind rebasing? You'll have to add ruby 3.1 (and uncomment 3.0) in the CI matrix [here](https://github.com/airbnb/ruby/blob/baee021e39a70924a973bb9a6b8c092b2d8afb9e/.github/workflows/rspec_rubocop.yml#L19-L33).

I just rebased my branch, but it seems that `First-time contributors need a maintainer to approve running workflows`

**Comment by pariser** [comment]
Approved your workflow! Tests are failing in 2.4 due to rubocop upgrade. That's a good tradeoff (support 3.1, drop support for 2.4). Feel free to update the matrix, and bump the `required_ruby_version` ([link](https://github.com/airbnb/ruby/blob/1939a2fb61db4e1f5cf7738e34d0c756f17f5afe/rubocop-airbnb/rubocop-airbnb.gemspec#L18)).

**Comment by RenzoMinelli** [comment]
@pariser seems that an approval in needed for each workflow 😅. I removed 2.4 in the last commit.

**Comment by pariser** [comment]
So it seems! I'll unlock CI right now

**Review by pariser — APPROVED** [review]
Thank you! I'll merge this and cut a new (major) release today.

**Comment by RenzoMinelli** [comment]
> Thank you! I'll merge this and cut a new (major) release today.

Nice! Thanks

**Pull request merged**
